doc/hardware/porting/board_porting.rst

```diff
@@ -242,6 +242,36 @@ followed by trial and error. If you want to understand details, you will need
 to read the rest of the devicetree documentation and the devicetree
 specification.
 
+Board extensions
+****************
+
+Board extensions are board fragments that may be present in any board root
+folder, under ``${BOARD_ROOT}/extensions/${BOARD}``. The board extension
+directory may contain Kconfig fragments and/or devicetree overlays. Board
+extensions are, by default, automatically loaded and applied on top of board
+files, before anything else. There is no guarantee on which order extensions are
+applied, in case multiple exist. This feature shall be disabled by passing
+``-DBOARD_EXTENSIONS=OFF`` to CMake.
+
+Board extensions are designed for downstream users, for example,
+``example-application`` or vendor SDKs. In some situations, certain hardware
+description or `choices <devicetree-chosen-nodes>`_ can not be added in the
+upstream Zephyr repository, but they do in a downstream project, where custom
+bindings or driver classes can also be created. This feature may also be useful
+in development phases, when the board skeleton lives upstream, but other
+features are developed in a downstream module.
+
+Note that board extensions need to follow the
+:ref:`same guidelines <porting-general-recommendations>` as regular boards. For
+example, it is wrong to enable extra peripherals or subsystems in a board
+extension.
+
+.. warning::
+
+   Board extensions are not allowed in any module hosted by the Zephyr project
+   organization. Any board changes are required to be submitted to the main
+   Zephyr repository.
```

Comment:
Above it says it can be used in example application which is part of the zephyr project organisation, so would be better if either example application was removed or an exclusion for that specific project was added here with a note that it would be to demonstrate example usage only

Reply:
extended docs, check now, thanks for pointing this out